### ISSUE TITLE
selective enabling of Alpheios on components

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios-core#96]
for alpheios-project/components#129 we disabled alpheios on the panel and popup . sometimes we want to be able to enable it selectively. This requires some thought about the best way to do it but, for example, the ge'ez parser provides short definitions for its words in Latin. We should able to enable Alpheios on those definitions in the popup.

One way to do this might be to look for the language code on the text that is displayed in a component and compare that to available languages to determine if Alpheios can be activated. But we might need even finer grained control of that.

Probably an issue for after refactoring of component state and data.

**Comment by balmas** [comment]
I need to write clear requirements for this before we can work on it.

**Comment by balmas** [comment]
I have done the following:

(1) added support for a new data attribute `data-alpheios-enabled="all"` . If this attribute is set in a child of an element which has the `data-alpheiosignore="all"` attribute set, then it will override the ignore instruction and enable alpheios for the element and its children. NB: this should be documented in the FAQ and embed-lib documentation. I added a documentation label to issue to remind us to do that...

(2) I have made a chance to the short definition component on the popup. If the definition is in a language that is supported by Alpheios (i.e one of the Alpheios target languages) then it will get this data attribute and Alpheios lookups will be enabled on the definition text. 

This can be tested with the Ge'ez word "ሀገርየ"" which has a definition in Latin.  

Additional issues can be entered to cover adding similar functionality on other elements of the Alpheios UI.

``

**Comment by balmas** [comment]
Alpheios Components 3.3.0-qa.20200501441

**Comment by monzug** [comment]
I tested the word ሀገርየ and I was able to lookup terra inhabitata and ager cultus. I have a question, as these lookups are initiated as child of a word in an other language, would it make more sense (or be more practical) to have the initial lookup on the background so when I quick look for the e.g. latin word patria, I can go back to my original word ሀገርየ?

**Comment by balmas** [comment]
> I tested the word ሀገርየ and I was able to lookup terra inhabitata and ager cultus. I have a question, as these lookups are initiated as child of a word in an other language, would it make more sense (or be more practical) to have the initial lookup on the background so when I quick look for the e.g. latin word patria, I can go back to my original word ሀገርየ?

It would but that is a fairly big change. I think we do have already a separate issue for supporting multiple popups #86 . This is a good use case for that issue.

**Comment by monzug** [comment]
an other one: I have tried on ὀδόντων. who has latin words in the short
def. I cannot lookup for dens, dentis.  is because short def is a mixed of
languages?

**Comment by monzug** [comment]
issue #367

**Comment by balmas** [comment]
> an other one: I have tried on ὀδόντων. who has latin words in the short def. I cannot lookup for dens, dentis. is because short def is a mixed of languages?

Yes, it's because that text (which shouldn't be there anyway) doesn't have the latin text identified as latin

**Comment by monzug** [comment]
so I can close this one as the only issue is covered by #86
